src/services/arweave/AntContract.tsx

```diff
@@ -0,0 +1,299 @@
+import {
+  ANTContractState,
+  ArweaveDataProvider,
+  ArweaveTransactionID,
+  SmartweaveDataProvider,
+} from '../../types';
+
+export class AntContract {
+  private _provider: SmartweaveDataProvider & ArweaveDataProvider;
+  state?: ANTContractState = undefined;
```

Comment:
this might be a good chance to use a `Map` given we will be writing/rewriting fields. i know warp doesn't play nice with raw maps, so we could introduce a `toJSON` function as well

Reply:
also, we could elevate the attributes of state to getters/setters on this class - what do you think?
```
class foo {
    private _bar: boolean = false;
    get bar(): boolean {
        return this._bar;
    }
    set bar(value: boolean) {
        this._bar = value;
    }
}
```